libparsec/crates/client/tests/unit/certificates_ops/store.rs

```diff
@@ -0,0 +1,393 @@
+
+    // Add new certificate
+    write
+        .add_next_certificate(1, &alice_update_certif, &alice_update_signed)
+        .await
+        .unwrap();
+
+    // Check that certificate is present
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, alice_update_certif);
+
+    store.stop().await;
+}
+
+#[parsec_test(testbed = "empty")]
+async fn add_sequester_authority_certificate(env: &TestbedEnv) {
+    let env = env.customize(|builder| {
+        builder
+            .bootstrap_organization("alice")
+            .and_set_sequestered_organization();
+    });
+    let alice = env.local_device("alice@dev1");
+    let store = certificates_store_factory(&env, &alice).await;
+
+    let (authority_certif, authority_signed) = env.get_sequester_authority_certificate();
+    let authority_certif = AnyArcCertificate::SequesterAuthority(authority_certif);
+
+    let write = store.for_write().await;
+
+    // Check that certificate is absent
+    let err = write.get_any_certificate(1).await.unwrap_err();
+    p_assert_matches!(err, GetCertificateError::NonExisting);
+
+    // Add new certificate
+    write
+        .add_next_certificate(1, &authority_certif, &authority_signed)
+        .await
+        .unwrap();
+
+    // Check that certificate is present
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, authority_certif);
+
+    store.stop().await;
+}
+
+// TODO: CertificatesStore doesn't work with SequesterServiceCertificate for
+// the moment, because it is signed with SequesterSigningKeyDer and it try to
+// unsecure_load with VerifyKey, which is wrong. But SequesterVerifyKey need to
+// know the size of the key, because the signature size depends on.
+#[ignore]
+#[parsec_test(testbed = "empty")]
+async fn add_sequester_service_certificate(env: &TestbedEnv) {
+    let env = env.customize(|builder| {
+        builder
+            .bootstrap_organization("alice")
+            .and_set_sequestered_organization();
+        builder.new_sequester_service();
+    });
+    let alice = env.local_device("alice@dev1");
+    let store = certificates_store_factory(&env, &alice).await;
+
+    let (service_certif, service_signed) = env.get_sequester_service_certificate(
+        SequesterServiceID::from_hex("00000000-0000-0000-0000-000000000001").unwrap(),
+    );
+    let service_certif = AnyArcCertificate::SequesterService(service_certif);
+
+    let write = store.for_write().await;
+
+    // Check that certificate is absent
+    let err = write.get_any_certificate(1).await.unwrap_err();
+    p_assert_matches!(err, GetCertificateError::NonExisting);
+
+    // Add new certificate
+    write
+        .add_next_certificate(1, &service_certif, &service_signed)
+        .await
+        .unwrap();
+
+    // Check that certificate is present
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, service_certif);
+
+    store.stop().await;
```

Comment:
retart the store and try to access the certificate (this ensure we don't rely on a cache but on data in persistent database)